Gateway/Request/Invoice/AbstractInvoiceDataBuilder.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Buckaroo\Magento2\Gateway\Request\Invoice;
+
+use Buckaroo\Magento2\Gateway\Request\AbstractDataBuilder;
+use Buckaroo\Magento2\Helper\Data as BuckarooHelper;
+
+abstract class AbstractInvoiceDataBuilder extends AbstractDataBuilder
+{
+    private BuckarooHelper $buckarooHelper;
+    protected int $numberOfInvoices;
+    protected bool $capturePartial = true;
+    protected float $currentInvoiceTotal;
+
+    public function __construct(BuckarooHelper $buckarooHelper)
+    {
+        $this->buckarooHelper = $buckarooHelper;
+    }
+
+    public function initialize(array $buildSubject): array
+    {
+        $data = parent::initialize($buildSubject);
+
+        $order = $this->getOrder();
+
+        $totalOrder = $order->getBaseGrandTotal();
+        $this->numberOfInvoices = $order->getInvoiceCollection()->count();
+        $this->currentInvoiceTotal = 0;
+
+        // loop through invoices to get the last one (=current invoice)
+        if ($this->numberOfInvoices) {
+            $oInvoiceCollection = $order->getInvoiceCollection();
```

Comment:
a collection usually has a `last()` method to get the last item in the collection, or use `end()`

Reply:
I used: $invoiceCollection->getLastItem() in both places